### PR TITLE
ci: use TinyGo version in artifact files

### PR DESCRIPTION
**Pull request by aykevl**
This avoids needing to rename them ourselves (which is kinda annoying) and also avoids mistakes in the process.

### CODE REVIEW

File: .github/workflows/tinygo-extract-version.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+# Extract the version string from the source code, to be stored in a variable.
+grep 'const version' goenv/version.go | sed 's/^const version = "\(.*\)"$/version=\1/g'
```
Comment (dgryski) [inline .github/workflows/tinygo-extract-version.sh:4]: grep not needed
```
sed -n '/^const version =/{ s/^const version = "\(.*\)"$/version=\1/; p;}' goenv/version.go
```

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -152,11 +157,11 @@ jobs:
       - name: Download release artifact
         uses: actions/download-artifact@v4
         with:
-          name: linux-amd64-double-zipped
+          name: linux-amd64-double-zipped-${{ needs.build-linux.outputs.version }}
       - name: Extract release tarball
         run: |
           mkdir -p ~/lib
-          tar -C ~/lib -xf tinygo.linux-amd64.tar.gz
+          tar -C ~/lib -xf tinygo*.linux-amd64.tar.gz
```
Comment (dgryski) [inline .github/workflows/linux.yml]: Any reason we're globbing here instead of using the `...outputs.version` variable?

File: .github/workflows/linux.yml [hunk 3]
```diff
@@ -297,6 +302,9 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+      - name: Get TinyGo version
+        id: version
+        run: ./.github/workflows/tinygo-extract-version.sh >> "$GITHUB_OUTPUT"
```
Comment (dgryski) [inline .github/workflows/linux.yml]: Above you used `tee -a` instead of `>>`.  Why the change?

**Review by dgryski — APPROVED** [review]
LGTM.

File: .github/workflows/tinygo-extract-version.sh (hunk 1 above)
Comment (aykevl) [reply .github/workflows/tinygo-extract-version.sh:4]: True, though I think the `grep` and `sed` together are more readable? This way the `sed` is just a simple regex search-and-replace instead of the more complicated code you wrote that I don't quite understand.
I can update if you think avoiding `grep` is better.

File: .github/workflows/linux.yml (hunk 3 above)
Comment (aykevl) [reply .github/workflows/linux.yml]: Fixed. This was the older code. I updated the code in most places (from `>>` to `tee -a` to be able to see the output of the step), but apparently missed this one.

File: .github/workflows/linux.yml (hunk 2 above)
Comment (aykevl) [reply .github/workflows/linux.yml]: No, it was just easier at the time. Fixed.

File: .github/workflows/tinygo-extract-version.sh (hunk 1 above)
Comment (dgryski) [reply .github/workflows/tinygo-extract-version.sh:4]: Nah, you can leave the grep if you prefer it.

Here's a breakdown of the `sed` code.

Sed, like awk, is can do `/pattern/ { action }`, so that's what we have here.  The `-n` flag tells `sed` not to print every line (the default), so  in our action after the replacement we need to explicitly `p` to print the current line.  If this was happening on a huge file I'd probably add a `q` to quit so `sed` didn't continue to process the entire file after it found the one match it was looking for

**Review by dgryski — APPROVED** [review]
LGTM

**Comment by dgryski** [comment]
Conflict that needs to be fixed before we can merge this.

**Comment by aykevl** [comment]
Fixed the conflict!
